src/v8/v8.cc

```diff
@@ -66,6 +68,14 @@ class V8 : public WasmVm {
             const std::unordered_map<uint32_t, std::string> &function_names) override;
   std::string_view getPrecompiledSectionName() override;
   bool link(std::string_view debug_name) override;
+  void terminateExecution() override {
+    auto *store_impl = reinterpret_cast<wasm::StoreImpl *>(store_.get());
+    auto *isolate = store_impl->isolate();
+    isolate->TerminateExecution();
+    while (isolate->IsExecutionTerminating()) {
+      std::this_thread::yield();
+    }
+  }
```

Comment:
Should we log this event? e.g.
```suggestion
    integration()->trace("[host->vm] Terminated");
  }
```

Reply:
If we don't mark it as failed, the user can may reuse it. If we mark it as fail explicitly, we don't need to wait for the termination to finish.

Reply:
It seems that fail() is not thread safe, but terminate() should be, so I get a tsan error. Do you have any suggestion?

Reply:
Let's drop explicit `fail()` call for now, since it's going to turn into a trap and failed WasmVM anyway when unrolling.